engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -371,6 +360,50 @@ class Print : public PhysicalPlan {
 	std::ostream * ofs = nullptr;
 	std::mutex print_lock;
 };
+ 
+
+class OutputKernel : public kernel {
+public:
+	OutputKernel() : kernel("OutputKernel") {  }
+	virtual kstatus run() {
+		output = std::move(this->input_.get_cache()->pullFromCache());
+		return kstatus::stop;
+	}
+	
+	frame_type	release() {
+		return std::move(output);
+	}
+
+protected:
+	frame_type output;
+};
+
+
+namespace test {
```

Comment:
can we move testing code to other files or was that arduous for some reason?

Reply:
all right